src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1052 @@
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: api-workload-ingress
+  annotations:
+    'spec.ingressClassName': "nginx"
+    ingress.kubernetes.io/ssl-redirect: "false"
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+spec:
+  rules:
+    - host: k8splayworkload.com
+      http:
+        paths:
+          - path: /
+            pathType: ImplementationSpecific
+            backend:
+              service:
+                name: workload-service
+                port:
+                  number: 9002
+
+---
+# workload auto-scaling strategy
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: workload-horizontal-autoscaling
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: workload-api
+  minReplicas: 1
+  maxReplicas: 4
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: 75
+    - type: Resource
+      resource:
+        name: memory
+        target:
+          type: Utilization
+          averageUtilization: 75
+  behavior:
+    scaleUp:
+      policies:
+        - periodSeconds: 60
+          type: Percent
+          value: 10
+      selectPolicy: Max
+      stabilizationWindowSeconds: 120
+    scaleDown:
+      policies:
+        - periodSeconds: 60
+          type: Percent
+          value: 1
+        - type: Pods
+          value: 1
+          periodSeconds: 60
+      selectPolicy: Max
+      stabilizationWindowSeconds: 120
+
+# Add environment variables
+```
+
+#### Auto-scaling
+We have added a horizontal auto-scaling policy for this deployment.
+* **minReplicas** is the lower limit for the number of replicas to which the auto-scaler can scale down. It defaults to 1 pod.
+* **maxReplicas** is the upper limit for the number of replicas to which the auto-scaler can scale up. It cannot be less that minReplicas.
+* **metrics** contains the specifications for which to use to calculate the desired replica count (the maximum replica count across all metrics will be used
+* **metrics.type** is the type of metric source
+* **metrics.type.resource**  refers to a resource metric (such as those specified in requests and limits) known to
+  Kubernetes describing each pod in the current scale target (e.g. CPU or memory)
+* **metrics.resource.name** is the name of the resource in question.
+* **metrics.resource.target** specifies the target value for the given metric.
+* **metrics.resource.target.type** represents whether the metric type is Utilization, Value, or AverageValue.
+* **metrics.resource.target.averageUtilization** is the target value of the average of the resource metric across all
+  relevant pods, represented as a percentage of the requested value of the resource for the pods. Currently only valid for Resource metric source type.
+
+with this bash script you can overload your scala play framework application running in the pod, and see how the
```

Comment:
- capitalize start of sentence
- Play Framework, Kubernetes

Reply:
Done

Reply:
Sentence changed